tensorflow_lite_support/c/task/vision/object_detector.cc

```diff
@@ -0,0 +1,203 @@
+}
+
+TfLiteObjectDetector* TfLiteObjectDetectorFromOptions(
+    const TfLiteObjectDetectorOptions* options, TfLiteSupportError** error) {
+  StatusOr<ObjectDetectorOptionsCpp> cpp_option_status =
+      CreateObjectDetectorCppOptionsFromCOptions(options);
+
+  if (!cpp_option_status.ok()) {
+    ::tflite::support::CreateTfLiteSupportErrorWithStatus(
+        cpp_option_status.status(), error);
+    return nullptr;
+  }
+
+  StatusOr<std::unique_ptr<ObjectDetectorCpp>> detector_status =
+      ObjectDetectorCpp::CreateFromOptions(cpp_option_status.value());
+
+  if (detector_status.ok()) {
+    return new TfLiteObjectDetector{.impl = std::move(detector_status.value())};
+  } else {
+    ::tflite::support::CreateTfLiteSupportErrorWithStatus(
+        detector_status.status(), error);
+    return nullptr;
+  }
+}
+
+TfLiteDetectionResult* GetDetectionResultCStruct(
+    const DetectionResultCpp& detection_result_cpp) {
+  auto c_detections =
+      new TfLiteDetection[detection_result_cpp.detections_size()];
+
+  for (int i = 0; i < detection_result_cpp.detections_size(); ++i) {
+    const DetectionCpp& detection = detection_result_cpp.detections(i);
+
+    auto c_categories = new TfLiteCategory[detection.classes_size()];
+    c_detections[i].size = detection.classes_size();
+
+    for (int j = 0; j < detection.classes_size(); ++j) {
+      const ClassCpp& classification = detection.classes(j);
+      c_categories[j].index = classification.index();
+      c_categories[j].score = classification.score();
+
+      if (classification.has_class_name())
+        c_categories[j].label = strdup(classification.class_name().c_str());
+      else
+        c_categories[j].label = nullptr;
+
+      if (classification.has_display_name())
+        c_categories[j].display_name =
+            strdup(classification.display_name().c_str());
+      else
+        c_categories[j].display_name = nullptr;
+    }
+    c_detections[i].categories = c_categories;
+  }
+
+  auto c_detection_result = new TfLiteDetectionResult;
+  c_detection_result->detections = c_detections;
+  c_detection_result->size = detection_result_cpp.detections_size();
+
+  return c_detection_result;
+}
+
+TfLiteDetectionResult* TfLiteObjectDetectorDetect(
+    const TfLiteObjectDetector* detector, const TfLiteFrameBuffer* frame_buffer,
+    TfLiteSupportError** error) {
+  if (detector == nullptr) {
+    tflite::support::CreateTfLiteSupportError(
+        kInvalidArgumentError, "Expected non null object detector.", error);
+    return nullptr;
+  }
+
+  StatusOr<std::unique_ptr<FrameBufferCpp>> cpp_frame_buffer_status =
+      ::tflite::task::vision::CreateCppFrameBuffer(frame_buffer);
+  if (!cpp_frame_buffer_status.ok()) {
+    tflite::support::CreateTfLiteSupportErrorWithStatus(
+        cpp_frame_buffer_status.status(), error);
+    return nullptr;
+  }
+
+  StatusOr<DetectionResultCpp> cpp_detection_result_status =
+      detector->impl->Detect(*std::move(cpp_frame_buffer_status.value()));
```

Comment:
Is `std::move` needed here?

Reply:
Not needed. Fixed in both image_classifier and object_detector